### PR TITLE
fix(broker): fixing interrupting event subprocesses

### LINKED ISSUE #3680: Workflow instance is stuck when an interrupting event subprocess is triggered
**Describe the bug**
When a workflow instance waits on a parallel gateway (joining multiple paths) and an **interrupting** event subprocess is triggered then the instance is stuck. The active element instances are terminated but the event subprocess is not activated.

**To Reproduce**

image

* deploy a workflow with a parallel gateway and an event subprocess (message/timer/error)
* create a workflow instance and wait until the instance enters at the parallel gateway
* trigger the event subprocess
* verify that the active element instances are terminated but the event subprocess is not activated

**Expected behavior**
The event subprocess is activated after all element instances are terminated.

**Environment:**
 - Zeebe Version: 0.22.0

### LINKED ISSUE #3681: Subscriptions of event subprocesses are not closed when one interrupting event subprocess is triggered
**Describe the bug**
When an interrupting event subprocess is triggered then the (message/timer) subscriptions of other event subprocesses are **not** closed. The other subscription can not be triggered anymore but they are still open.

**To Reproduce**

image

* deploy a workflow with multiple event subprocesses - at least one interrupting 
* create a workflow instance 
* trigger the interrupting event subprocess
* verify that the other subscriptions are not closed

**Expected behavior**
All subscriptions are closed after an interrupting event subprocess is triggered

**Environment:**
 - Zeebe Version: 0.22.0

### PR DESCRIPTION
**Pull request by saig0**
## Description

* when an interrupting event subprocess is triggered then terminate the child instances and consume waiting tokens (e.g. on a joining parallel gateway)
* close event subscriptions when an interrupting event subprocess is triggered

## Related issues

closes #3680 
closes #3681 

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

File: engine/src/main/java/io/zeebe/engine/processor/workflow/handlers/eventsubproc/EventSubProcessEventOccurredHandler.java [hunk 1]
```diff
@@ -86,21 +94,28 @@ private long handleInterrupting(
...
+
+        terminatedChildInstances += 1;
       }
     }
 
-    return waitForTermination;
+    // consume all other active tokens (e.g. tokens waiting at a joining gateway)
+    final int zombies =
+        context.getFlowScopeInstance().getNumberOfActiveTokens() - terminatedChildInstances;
+    for (int z = 0; z < zombies; z++) {
```
Comment (MiguelPires) [inline engine/src/main/java/io/zeebe/engine/processor/workflow/handlers/eventsubproc/EventSubProcessEventOccurredHandler.java]: Just a minor nit pick: This was a little odd to me, I guess I'm so used to seeing a `for` when iterating, that I had to read the next line again looking for the `z` variable. It also seems that we can achieve the same without it, like:

```suggestion
    while (zombies-- > 0) {
```
or
```suggestion
while (zombies > 0) {
zombies--;
...
```

**Review by miguelpires — APPROVED** [review]
Looks good. Just had one nitpick that you may or may not agree with

File: engine/src/main/java/io/zeebe/engine/processor/workflow/handlers/eventsubproc/EventSubProcessEventOccurredHandler.java (hunk 1 above)
Comment (saig0) [reply engine/src/main/java/io/zeebe/engine/processor/workflow/handlers/eventsubproc/EventSubProcessEventOccurredHandler.java]: Thanks @MiguelPires. I see your point. But as for now, I will not change it. There are 2 other places which almost the same code :see_no_evil: I hope to get rid of it soon.

**Pull request merged**